EIPS/eip-FOCIL.md

```diff
@@ -42,12 +42,17 @@ IL committee members construct their ILs and broadcast them over the P2P network
 
   By default, ILs are built by selecting raw transactions from the public mempool, ordered by priority fees, up to the IL’s maximum size in bits (e.g., 8 KB per IL). Additional rules can be optionally applied to maximize censorship resistance, such as prioritizing valid transactions that have been pending in the mempool the longest.
 
-#### Nodes
+#### Validators
+- **`Slot N`, `t=0 to 9s`**: Validators receive ILs from the P2P network and store (1) all new ILs that pass the CL P2P validation rules, and any evidence of IL equivocation by committee members (i.e., if multiple ILs are received from the same committee member).
 
-- **`Slot N`, `t=0 to 9s`**:
-Nodes receive ILs from the P2P network and only forward and cache those that pass the CL P2P validation rules.
+- **`Slot N`, `t=9s`**: View freeze deadline. At this point, validators freeze their IL view and stop caching new ILs in memory.
 
-- **`Slot N`, `t=9s`**: IL freeze deadline. At this point, nodes freeze their IL view and stop caching new ILs in memory. After the deadline, nodes continue forwarding ILs to peers following the CL P2P validation rules, but they discard any new local ILs received after the deadline (`t=9s`) and keep only minimal information (i.e., the number of ILs forwarded per committee member) to forward according to the rules.
+- **`Slot N`, `t=9s` to `Slot N+1`, `t=4s`**: After the view freeze deadline, validators:
+  1. Do not store new ILs received after the deadline.
+  2. Continue forwarding ILs to peers following the CL P2P validation rules.
+  3. Record any evidence of IL equivocation that occurs after the view freeze deadline.
+
+After the attestation deadline of **`Slot N+1`, `t=4s`**, validators ignore any new ILs related to the previous slot's IL committee, and stop recording equivocation evidence for the previous slot's ILs.
 
 #### Block Producer
 - **`Slot N`, `t=0 to 11s`**: The block producer (i.e., a proposer or a proposer builder pair) receive ILs from the P2P network, forwarding and caching those that pass the CL P2P validation rules. Optionally, an RPC endpoint can be added to allow the block producer to request missing ILs from its peers (e.g., by committee index at `t=10s`).
```

Comment:
Proposer or proposer builder pair is a bit unclear to me. What is a proposer builder pair? If it's meant to be a local block builder (a proposer who also builds), this is confusing because then "just a proposer" would mean a proposer using mev-boost, and a proposer using mev-boost does not care about ILs at all since it does not build the payload. If instead a proposer builder pair is a proposer using mev-boost while a proposer is a local block builder, I think it's also confusing because in the "proposer builder pair" it is only actually the builder that deals with ILs. 

I would rephrase it to "a proposer doing local block building or an external builder", and perhaps change "block producer" just to builder, since here we are actually talking about the "payload producer", not the beacon block producer

Reply:
Both for clarity purposes but also because it's actually a strength of this mechanism that it does not require (non self building) beacon block proposers to do anything

Reply:
agreed, made the following changes for clarity: 

`#### Builder
- **`Slot N`, `t=0 to 11s`**: The builder (i.e., a proposer doing local block building or an external builder) receive ILs from the P2P network, forwarding and caching those that pass the CL P2P validation rules. Optionally, an RPC endpoint can be added to allow the builder to request missing ILs from its peers (e.g., by committee index at `t=10s`).

- **`Slot N`, `t=11s`**:
The builder freezes its view of ILs and asks the EL to update its execution payload by adding transactions from its view (the exact timings will be defined after running some tests/benchmarks).

#### Proposer
- **`Slot N+1`, `t=0s`**:
The proposer broadcasts its block with the up-to-date execution payload satisfying IL transactions over the P2P network.`